strax/context.py

```diff
@@ -1844,30 +1844,89 @@ def get_meta(self, run_id, target) -> dict:
 
     get_metadata = get_meta
 
-    def compare_metadata(self, run_id, target, old_metadata):
+    def compare_metadata(self, data1, data2, return_results=False):
         """Compare the metadata between two strax data.
 
-        :param run_id: run id to get
-        :param target: data type to get
-        :param old_metadata: path to metadata to compare, or a dictionary, or a tuple with another
-            run_id, target to compare against the metadata of the first id-target pair
+        :param data1, data2: either a list (tuple) of runid + target pair, or path to metadata to
```

Comment:
Can you add into your doc string the examples you provided here in github. Then I think it becomes more clea, like:

```python
"""
doc-string so far

    Examples:
    example code 1
    example code 2
"""
```